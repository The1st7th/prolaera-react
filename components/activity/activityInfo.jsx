import React from 'react';
import { Box, Item } from 'react-html-email';

class ActivityInfo extends React.Component {
  render() {
    return (
      <div
        style={{
          display: 'block',
          marginLeft: 'auto',
          marginRight: 'auto',
          marginBottom: '20px',
          textAlign: 'center',
          border: '1px solid lightgray',
          borderRadius: '5px',
          maxWidth: '500px',
          padding: '20px',
          fontFamily: 'Arial, sans-serif',
          boxShadow: '1px 1px 8px darkgray'
        }}
      >
        <Box align="center" width="100%">
          <Item />
        </Box>
      </div>
    );
  }
}

export default ActivityInfo;
