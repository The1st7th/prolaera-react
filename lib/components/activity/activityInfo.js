'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _createClass = (function() {
  function defineProperties(target, props) {
    for (var i = 0; i < props.length; i++) {
      var descriptor = props[i];
      descriptor.enumerable = descriptor.enumerable || false;
      descriptor.configurable = true;
      if ('value' in descriptor) descriptor.writable = true;
      Object.defineProperty(target, descriptor.key, descriptor);
    }
  }
  return function(Constructor, protoProps, staticProps) {
    if (protoProps) defineProperties(Constructor.prototype, protoProps);
    if (staticProps) defineProperties(Constructor, staticProps);
    return Constructor;
  };
})();

var _react = require('../../../../../Library/Caches/typescript/2.9/node_modules/@types/react');

var _react2 = _interopRequireDefault(_react);

var _reactHtmlEmail = require('react-html-email');

var _button = require('../button');

var _button2 = _interopRequireDefault(_button);

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

function _classCallCheck(instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError('Cannot call a class as a function');
  }
}

function _possibleConstructorReturn(self, call) {
  if (!self) {
    throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  }
  return call && (typeof call === 'object' || typeof call === 'function') ? call : self;
}

function _inherits(subClass, superClass) {
  if (typeof superClass !== 'function' && superClass !== null) {
    throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass);
  }
  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: { value: subClass, enumerable: false, writable: true, configurable: true }
  });
  if (superClass)
    Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : (subClass.__proto__ = superClass);
}

var ActivityInfo = (function(_React$Component) {
  _inherits(ActivityInfo, _React$Component);

  function ActivityInfo() {
    _classCallCheck(this, ActivityInfo);

    return _possibleConstructorReturn(
      this,
      (ActivityInfo.__proto__ || Object.getPrototypeOf(ActivityInfo)).apply(this, arguments)
    );
  }

  _createClass(ActivityInfo, [
    {
      key: 'render',
      value: function render() {
        var _props = this.props,
          _props$name = _props.name,
          name = _props$name === undefined ? '' : _props$name,
          _props$author = _props.author,
          author = _props$author === undefined ? '' : _props$author,
          profile_uid = _props.profile_uid,
          activity_id = _props.activity_id,
          _props$header = _props.header,
          header = _props$header === undefined ? 'You have been assigned to a new Activity:' : _props$header,
          _props$buttonText = _props.buttonText,
          buttonText = _props$buttonText === undefined ? 'View Details' : _props$buttonText,
          _props$button1Link = _props.button1Link,
          button1Link =
            _props$button1Link === undefined
              ? 'https://app.prolaera.com/#/users/' + ('' + profile_uid) + '/activities' + ('' + activity_id)
              : _props$button1Link;

        return _react2.default.createElement(
          'div',
          {
            style: {
              display: 'block',
              marginLeft: 'auto',
              marginRight: 'auto',
              marginBottom: '20px',
              textAlign: 'center',
              borderRadius: '5px',
              maxWidth: '500px',
              padding: '20px',
              fontFamily: 'Arial, sans-serif',
              backgroundColor: 'white'
            }
          },
          _react2.default.createElement(
            _reactHtmlEmail.Box,
            { align: 'center', width: '100%' },
            _react2.default.createElement(
              _reactHtmlEmail.Item,
              null,
              _react2.default.createElement('h4', null, header),
              _react2.default.createElement(
                'div',
                {
                  style: {
                    marginLeft: 'auto',
                    marginRight: 'auto',
                    marginBottom: '15px',
                    border: '1px solid lightgray',
                    boxShadow: '1px 1px 8px darkgray',
                    maxWidth: '300px'
                  }
                },
                _react2.default.createElement(
                  'h3',
                  { style: { marginBottom: '0px' } },
                  _react2.default.createElement('em', null, name)
                ),
                _react2.default.createElement(
                  'h4',
                  { style: { marginTop: '7px' } },
                  'By: ',
                  _react2.default.createElement('span', { style: { fontWeight: 'normal' } }, author)
                )
              )
            ),
            _react2.default.createElement(
              _reactHtmlEmail.Item,
              null,
              _react2.default.createElement(_button2.default, { text: buttonText, link: button1Link })
            )
          )
        );
      }
    }
  ]);

  return ActivityInfo;
})(_react2.default.Component);

exports.default = ActivityInfo;
