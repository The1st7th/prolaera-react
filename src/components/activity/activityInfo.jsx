import React from 'react';
import { Box, Item } from 'react-html-email';
// import Button from '../button';
import InfoHeader from '../infoHeader';

class ActivityInfo extends React.Component {
  render() {
    const {
      name = '',
      author = '',
      profile_uid,
      activity_id,
      buttonText = 'View Details',
      button1Link = 'https://app.prolaera.com/#/users/' + `${profile_uid}` + '/activities' + `${activity_id}`
    } = this.props;

    return (
      <Box align="center" className="container">
        <Item>
          <Box className="row">
            <tr>
              <th className="small-12 large-12 columns first last">
                <table align="center">
                  <tbody>
                    <tr>
                      <th>
                        <InfoHeader info={name} />
                        <p className="text-center small-text-center">By: {author}</p>
                      </th>
                    </tr>
                  </tbody>
                </table>
              </th>
            </tr>
          </Box>
        </Item>
        <Item>
          <Box className="row">
            <tr>
              <th className="small-12 large-12 columns first last">
                <table align="center">
                  <tbody>
                    <tr>
                      <th>
                        <center data-parsed="">
                          <table align="center" className="" style={{}}>
                            <tr>
                              <td>
                                <table>
                                  <tr>
                                    <td
                                      align="center"
                                      style={{
                                        padding: '18px',
                                        fontSize: '14pt',
                                        backgroundColor: '#72C02C',
                                        border: '1px solid #72C02C',
                                        borderRadius: '3px'
                                      }}
                                      className="text-center small-text-center"
                                    >
                                      <a
                                        align="center"
                                        style={{ color: '#FFFFFF', textDecoration: 'none' }}
                                        href={button1Link}
                                      >
                                        {buttonText}
                                      </a>
                                      {/* <Button text={buttonText} link={button1Link} /> */}
                                    </td>
                                  </tr>
                                </table>
                              </td>
                            </tr>
                          </table>
                        </center>
                      </th>
                    </tr>
                  </tbody>
                </table>
              </th>
            </tr>
          </Box>
        </Item>
      </Box>
    );
  }
}

export default ActivityInfo;
