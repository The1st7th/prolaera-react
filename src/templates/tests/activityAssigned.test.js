import renderer from 'react-test-renderer';
import writeFile from '../../helpers/writeFileHelper';
import activityAssignedEmail from '../builders/activityAssignedBuilder';
import activity from '../tests/json/completeActivity.json';
import user from '../tests/json/completeUser.json';
import inlineCss from 'inline-css';
var request = require('request');

const _sendEmail = (template, emails) => {
  return new Promise((resolve, reject) => {
    request(
      {
        url: 'https://api-dev.prolaera.com/v1/mailer',
        method: 'PUT',
        json: {
          template,
          emails
        }
      },
      (error, response, body) => {
        if (error) {
          console.error(error);
          return reject(error);
        }
        resolve({
          response,
          body
        });
      }
    );
  });
};

describe('Activity Assigned email', () => {
  const logoUrl = 'https://assets.prolaera.com/prolaeraLogo_fullText.png';

  it('checks snapshot', async () => {
    const emailHtml = await activityAssignedEmail(activity, user, logoUrl);
    const emailComponent = renderer.create(emailHtml);
    const activityJson = emailComponent.toJSON();
    expect(activityJson).toMatchSnapshot();
  });

  it('writes an html file', async () => {
    const email = await activityAssignedEmail(activity, user, logoUrl);
    const newEmail = await inlineCss(email, {
      url: ' ',
      preserveMediaQueries: true,
      applyWidthAttributes: true,
      applyTableAttributes: true
    });
    // await _sendEmail(newEmail, ['aflupton@gmail.com', 'andrew@prolaera.onmicrosoft.com', 'teagan@prolaera.com']);
    const saved = await writeFile(newEmail, 'activityTest.html');
    expect(saved).toEqual(true);
  });
});
