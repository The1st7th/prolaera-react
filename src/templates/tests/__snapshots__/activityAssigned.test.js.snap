// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity Assigned email checks snapshot 1`] = `
"<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Strict//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\\"><html lang=\\"en\\" xmlns=\\"http://www.w3.org/1999/xhtml\\"><head><meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=utf-8\\"/><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"/><title>You have been Assigned</title><style type=\\"text/css\\">@font-face {
	font-family: source-sans-pro;
	url(&#x27;https://fonts.googleapis.com/css?family=Source+Sans+Pro&#x27;);

}
html {
	font-family: source-sans-pro, sans-serif;
}

.textMuted {
	color: #333333;
}

#subHeaderWrapper {
	padding-top: 20px;
	padding-bottom: 0px;
}

.innerHtmlStyles {
	margin-top: 10px;
}

.innerHtmlStyles p, .innerHtmlStyles ol, .innerHtmlStyles ul  {
	margin-top: 10px;
}

button.buttonWidths {
	-webkit-appearance: none;
	-moz-appearance: none;
}

@media
only screen and (max-width: 768px)  {
	html {
		font-family: source-sans-pro, sans-serif;
	}

	body{
		background-color: #F7F7F7;
		margin-left: auto;
		margin-right: auto;
	}

	

	/* Force table to not be like tables anymore */
	table, thead, tbody, th, td, tr {
		display: block;		
	}
	
	#courseCertificate {
		max-width: 100%;
	}

	.courseCertificate {
		width: 100%;
		margin: auto;
	}
	
	.emailHeader td {
		width: 100%;
	}

	.emailHeader td img {
		margin-left: auto;
		margin-right: auto;
	} 

	.emailFooter {
		width:500px
		color: #F7F7F7;
		margin-right: auto;
		margin-left: auto;
		font-size: 10pt;
	}

	.emailFooter td {
	
		color: #F7F7F7;
		margin-right: auto;
		margin-left: auto;
		font-size: 10pt;
		
	}

	.report td {
		/* Behave  like a &quot;row&quot; */
		border: none;
		border-bottom: 1px solid #eee;
    	position: relative;
    	text-align: top;
	}

	.report td:before {
		/* Now like a table header */
		position: absolute;
		/* Top/left values mimic padding */
		top: 6px;
		left: 6px;
		width: 45%;
		padding-right: 10px;
		white-space: nowrap;
	}
	
	.customButton:hover {
		color:black;
		
	}
	#div1{
		text-align:center;
	}
	#one{
		
		position:relative;
		top:5px;
		paddingTop: 5px;
		height:23px;
		width:100px
	}
	#div2{
		text-align:center;
		padding-top:7px;
	}
}
	@media (min-width: 769px) {

		html{
			padding-left:25%;
			font-family: source-sans-pro, sans-serif;
		}
		body{
			background-color: #F7F7F7;
			margin-left: auto;
			margin-right: auto;
			max-width:55%;
			
		}
		/* Force table to not be like tables anymore */
		table, thead, tbody, th, td, tr {
			// display: block;		
		}

		#courseCertificate {
			max-width: 100%;
		}
		.courseCertificate {
			width: 100%;
			margin: auto;
		}
		
		.emailHeader td {
			
		}
	
		.emailHeader td img {
			margin-left: auto;
			margin-right: auto;
		} 
	
		.emailFooter {
			padding-top:10px;
			padding-bottom:10px;
			max-height:100px;
			width:100%;
			color: #F7F7F7;
			margin-right: auto;
			margin-left: auto;
			font-size: 10pt;
		}

		.emailFooter td {
			width:100%;
			color: #F7F7F7;
			margin-right: auto;
			margin-left: auto;
			font-size: 10pt;
			
		}

	
		.report td {
			/* Behave  like a &quot;row&quot; */
			border: none;
			border-bottom: 1px solid #eee;
			position: relative;
			text-align: top;
		}
	
		.report td:before {
			/* Now like a table header */
			position: absolute;
			/* Top/left values mimic padding */
			top: 6px;
			left: 6px;
			width: 45%;
			padding-right: 10px;
			white-space: nowrap;
		}
	
		.customButton:hover {
			color:black;
		
	  }
    #div1 {
		float:left;
		margin-left:5%;
	}
	#span1{
		float:left;
		padding:10px 5px 10px 10px;
	}
	#div2{
		text-align:right;
		padding:10px 10px 10px 10px;
		margin-left:5%
	}
	#one{
		paddingTop: 5px;
		height:23px;
		width:100px;
		position:relative;
		top:5px;
	}
	#footerId{
		padding-top:20px;
		padding-bottom:0px;
	}
	
/*End of media query*/
}</style></head><body style=\\"width:100%;margin:0;padding:0;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%\\"><table width=\\"100%\\" height=\\"100%\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" align=\\"left\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\" valign=\\"top\\"><table width=\\"600\\" align=\\"center\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><table class=\\"emailHeader\\" align=\\"center\\" width=\\"100%\\" style=\\"background-color:#F7F7F7\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\"><table align=\\"center\\" cellSpacing=\\"0\\" style=\\"padding:15px;max-width:584px;margin-top:25px;margin-bottom:20px\\"><tr align=\\"center\\"><td width=\\"292px\\"><img alt=\\"logo\\" src=\\"https://assets.prolaera.com/prolaeraLogo_fullText.png\\" width=\\"199.15254237288136\\" height=\\"47\\" style=\\"display:block;outline:none;border:none;text-decoration:none\\"/></td></tr></table></td></tr></tbody></table><div id=\\"subHeaderWrapper\\" style=\\"background-color:#FFFFFF;margin-left:auto;margin-right:auto;font-style:normal;color:black;font-weight:500;display:block;letter-spacing:1pt\\"><table class=\\"subHeader\\" align=\\"center\\" width=\\"100%\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td><table id=\\"subHeaderTable\\" style=\\"text-align:center;margin-left:auto;margin-right:auto\\"><tr><td><p style=\\"font-size:18pt;margin-bottom:5px\\">You have been</p></td></tr><tr><td><p style=\\"font-size:28px;margin-top:0px;margin-bottom:0px\\">Assigned</p></td></tr><tr style=\\"margin:auto;padding:0px 10px 10px 10px\\"><td style=\\"width:100%\\" align=\\"center\\"><div style=\\"max-width:85%;letter-spacing:1pt;padding:20px 20px 0px 20px\\">A new Activity</div><div style=\\"max-width:85%\\"><div></div></div></td></tr></table></td></tr></tbody></table></div><div style=\\"display:block;margin:auto;text-align:center;font-style:normal;font-weight:400;background-color:#FFFFFF\\"><table align=\\"center\\" width=\\"100%\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\"><div style=\\"width:90%;margin:auto\\"><div><table class=\\"courseReview\\" align=\\"center\\" width=\\"100%\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\" style=\\"padding:20px\\"><div style=\\"width:100%;height:20px;border-bottom:1px solid grey;text-align:center;box-sizing:border-box;position:relative;margin:auto;padding:10px\\"><span style=\\"font-size:14pt;background-color:#FFFFFF;padding:0px 20px;box-sizing:border-box;margin:0\\">Eloquent JavaScript Information</span></div></td></tr></tbody></table></div><p style=\\"padding-top:15px;margin-top:0px\\">By: Teagan Wrest</p></div></td></tr><tr><td align=\\"center\\"><div style=\\"display:inline-block;margin:auto;padding:20px\\"><div class=\\"customButton\\" align=\\"center\\" style=\\"text-align:center;height:100%;width:100%\\"><a href=\\"https://app.prolaera.com/#/users/e7cc804d-8ab1-431e-96a6-232d477fb7e0/activitiesf3fe0588-8c4e-4f87-b18f-fa90dfa61db7\\"><button class=\\"buttonWidths\\" style=\\"height:100%;padding:14px 14px 14px 14px;color:#FFFFFF;font-size:10pt;width:100%;text-decoration:none;border-radius:3px;border-style:none;border-color:#72C02C;background-color:#72C02C\\">View Details</button></a></div></div></td></tr></tbody></table></div><div><table id=\\"footerId\\" class=\\"emailFooter\\" align=\\"center\\" cellSpacing=\\"20\\" style=\\"background-color:#333333;padding-top:5px\\" height=\\"60\\" cellPadding=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td><div id=\\"div1\\"><span id=\\"span1\\" style=\\"color:#F7F7F7;font-size:10pt;font-weight:100;letter-spacing:1pt\\">Powered By </span><img id=\\"one\\" src=\\"https://gallery.mailchimp.com/eb930e30a15249a18d658c183/images/196fa069-3113-464c-a304-a1aa9f99fe22.png\\" alt=\\"react\\"/></div><div id=\\"div2\\" style=\\"color:#F7F7F7;font-size:10pt;font-weight:100;letter-spacing:1pt\\">Â© Prolaera 2018</div></td></tr></tbody></table></div></tbody></table></td></tr></tbody></table></body></html>"
`;
